app/controllers/system_admin/funding/trainee_summaries_controller.rb

```diff
@@ -32,11 +32,20 @@ def redirect_to_lead_school_path
           redirect_to(provider_funding_trainee_summary_path(provider_id: organisation.id, academic_year: selected_academic_cycle.start_year))
         when School
           redirect_to(lead_school_funding_trainee_summary_path(lead_school_id: organisation.id, academic_year: selected_academic_cycle.start_year))
+        when LeadPartner
+          redirect_to(lead_partner_funding_trainee_summary_path(lead_partner_id: organisation.id, academic_year: selected_academic_cycle.start_year))
         end
       end
 
       def organisation
-        @organisation ||= params[:provider_id].present? ? Provider.find(params[:provider_id]) : School.find(params[:lead_school_id])
+        @organisation ||=
+          if params[:provider_id].present?
+            Provider.find(params[:provider_id])
+          elsif params[:lead_school_id].present?
+            School.find(params[:lead_school_id])
+          else
+            LeadPartner.find(params[:lead_partner_id])
+          end
       end
```

Comment:
As this method is been repeated in `PaymentSchedulesController` I suggest to move it to `::Funding::BaseFundingController`